lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -670,20 +670,44 @@ public static function clean_cached_data() {
 	}
 
 	/**
-	 * Returns the style variations defined by the theme.
+	 * Returns an array of all nested json files within a given directory.
 	 *
-	 * @since 6.0.0
+	 * @since 6.2.0
+	 *
+	 * @param dir $dir The directory to recursively iterate and list files of.
+	 * @return array The merged array.
+	 */
+	private static function recursively_iterate_JSON( $dir ) {
+		$nested_files      = new RecursiveIteratorIterator( new RecursiveDirectoryIterator( $dir ) );
+		$nested_json_files = iterator_to_array( new RegexIterator( $nested_files, '/^.+\.json$/i', RecursiveRegexIterator::GET_MATCH ) );
+		return $nested_json_files;
+	}
+
+	/**
+	 * Returns the style variations defined by the theme (parent and child).
 	 *
 	 * @return array
 	 */
 	public static function get_style_variations() {
-		$variations     = array();
-		$base_directory = get_stylesheet_directory() . '/styles';
+		$variations         = array();
+		$base_directory     = get_stylesheet_directory() . '/styles';
+		$template_directory = get_template_directory() . '/styles';
 		if ( is_dir( $base_directory ) ) {
-			$nested_files      = new RecursiveIteratorIterator( new RecursiveDirectoryIterator( $base_directory ) );
-			$nested_html_files = iterator_to_array( new RegexIterator( $nested_files, '/^.+\.json$/i', RecursiveRegexIterator::GET_MATCH ) );
-			ksort( $nested_html_files );
-			foreach ( $nested_html_files as $path => $file ) {
+			$variation_files = static::recursively_iterate_JSON( $base_directory );
+			if ( $template_directory !== $base_directory && is_dir( $template_directory ) ) {
```

Comment:
It fails to show style variations from parent when child theme doesn't have a `styles` directory. I guess, it should still have them displayed.

Reply:
Nice catch, thanks for testing.

I addressed this in https://github.com/WordPress/gutenberg/pull/46554/commits/1fdb35d970293542376e0a69d076a10fb6c47f65